lib/lte_link_control/lte_lc.c

```diff
@@ -138,7 +138,12 @@ K_WORK_DEFINE(lte_lc_edrx_ptw_send_work, lte_lc_edrx_ptw_send_work_fn);
 static void lte_lc_edrx_req_work_fn(struct k_work *work_item);
 K_WORK_DEFINE(lte_lc_edrx_req_work, lte_lc_edrx_req_work_fn);
 
-K_THREAD_STACK_DEFINE(lte_lc_work_q_stack, 1024);
+#if defined(CONFIG_LOG_MODE_IMMEDIATE)
+#define LTE_LC_WORKQUEUE_STACK_SIZE 1152
```

Comment:
Looks like for my build it needs at least 1200 bytes with LTE_LINK_CONTROL_LOG_LEVEL_DBG.

Reply:
Can you also try what does it need with `CONFIG_LOG_MODE_DEFERRED`? I'm curious because with modem_shell sample the highest I've seen are 736 / 1064 bytes.
What I missed looking at the code was that we are calling event callback from this work queue. So I'll need to make Kconfig option rather than hard coded value.

Reply:
I've increased the default with `CONFIG_LOG_MODE_IMMEDIATE` to 1280.

Reply:
With deferred logging the stack uses 576 bytes for me. I have not tested with any PSM or EDRX stuff enabled.

Reply:
Ok, I'll run some more tests with different SIM cards before I get this merged.